### DISCUSSION TITLE
Related to the "Delta Filter" but kept apart for readability

### DISCUSSION THREAD

**Discussion by luissimoes** [dbuezas/lovelace-plotly-graph-card#378] [Q&A]
I analysed alternatives eg instead of Delta and Sum I used state... 
But there is also a problem because the sensor sometimes resets 10 seconds before 00:00 and other times 10 seconds *after** 00:00

How can I tell plotly to get always the state for say 30 seconds **before** now?

I tried 
- map_x: x - 30* 1000 but not success

Here the case "before"
<img width="1128" alt="Pasted Graphic 22" src="https://github.com/dbuezas/lovelace-plotly-graph-card/assets/42578174/0d013a79-a1ae-47bb-baec-be2b2e06ed04">

**Comment by dbuezas** [comment]
Not sure what your objective is. Can you clarify?

**Reply by luissimoes** [reply]
hi
sorry if I was not clear.
this is in part related to the post on the "delta behaviour"...
From the above graph we see that the sensor (its a template as explained in the other post) resets short before 00:00.....

This is a problem when using STATE as statistic and period:day because the last state will be zero... --> wrong results.

To avoid this, since sometimes the sensor resets *after* 00:00 (as mentioned in the other post, i.e. exactly the symmetrical behaviour relative to 00:00) an alternative would be to use the value say 20 seconds before 00:00
- so I am looking to a way to force a sampling so that it is never at the end of the day but max (say) 20 seconds before

**Reply by dbuezas** [reply]
Oh, I see. Either map_x before resample or time_offset should both work, but you need to add to x, not subtract

**Comment by luissimoes** [comment]
Ok 🙏 
I still have the discrepancy I mentioned in the previous post (on delta filter) but I will try this.
It seems it only shifts the values…. Doesn’t really resemble them (using offset)

**Reply by dbuezas** [reply]
What do you mean with "resemble"?

**Reply by luissimoes** [reply]
Resample 😀

**Reply by dbuezas** [reply]
Ohhhh i see. Well, add a resample at the end!
